ansible_collections/arista/avd/plugins/action/topology_generator.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+import glob
+from ansible.errors import AnsibleActionFail
+from ansible.plugins.action import ActionBase
+import ansible_collections.arista.avd.plugins.plugin_utils.topology_generator_utils as gt
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = {}
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+        if self._task.args and "structured_config" not in self._task.args:
+            raise AnsibleActionFail("Missing 'structured_config' variable.")
+        path = self._task.args["structured_config"]
+        self.driver_func(path)
+        return result
+
+
+    def driver_func(self, directory_path):
+        graph_dict = {}
+        json_input = []
+        finaldict = {}
+        files = glob.glob(directory_path+"/*.yml")
+
+        output_list = []
+        for index, file in enumerate(files):
+            data = gt.read_yaml_file(file)
+            node_dict = gt.create_node_dict(data, file)
+            output_list = gt.structured_config_to_topology_input(
+                output_list, node_dict, data['diagram_groups'], current_dict={})
+        root_dict = gt.find_root_nodes(output_list[0])
+        nodes = []
+        output_list[0]["nodes"].append(root_dict)
+        global_node_list, graph_dict = gt.create_graph_dict(output_list)
+        nodes_rank_dict = gt.find_node_levels(graph_dict, '0', global_node_list)
+        rank_nodes_list = []
+        for k, v in nodes_rank_dict.items():
+            rank_nodes_list = rank_nodes_list + v
```

Comment:
```suggestion
        rank_nodes_list = [v for _, v in nodes_rank_dict.items()]
```

Reply:
I won't work, we need list of values, but above solution returning list of list.

Reply:
Ah yeah my bad..

so in that case please use one of these
```
        for v in nodes_rank_dict.values():
            rank_nodes_list += v
```

or

```
        for v in nodes_rank_dict.values:
            rank_nodes_list.extend(v)
```